docker/Dockerfile

```diff
@@ -35,6 +35,8 @@ RUN set -ex && \
     useradd  --uid=${celeborn_uid} --gid=${celeborn_gid} celeborn -d /home/celeborn -m && \
     mkdir -p ${CELEBORN_HOME}
 
+ENV LD_PRELOAD=/usr/lib/x86_64-linux-gnu/libjemalloc.so
```

Comment:
From my experience, I think we could use code to switch the environment via `uname -m` (we can refer to the [flink-docker code](https://github.com/apache/flink-docker/blob/45c6d230407d89aa83b0d170dd056d6868cf808e/1.17/scala_2.12-java8-ubuntu/docker-entrypoint.sh#L92C3-L109)). 

I think it make sense to make this option configurable and we shall enable this by default.

Reply:
`uname -m` would be a good idea, but does it work in ENV command? 

Also, is it only beneficial inside the container? if not, we'd better supply a general solution. e.g. port the jemalloc detection and enabling logic into the bootstrap shell script, and introduce an env var like CELEBORN_DISABLE_JEMALLOC to make it configurable.

Then, the behavior is consistent inside and outside of the container: 
- enable jemalloc by default when libjemalloc.so is available
- user can set CELEBORN_DISABLE_JEMALLOC=1 to explicitly disable the jemalloc usage

thought?

Reply:
+1. I think your proposal is almost the same as what Flink does currently. The `uname -m` would be executed in the `docker-entrypoint.sh` instead of `Dockerfile` building.

Reply:
To be clear, what I mean "bootstrap shell script" is `sbin/load-celeborn-env.sh`(or maybe anther one), not `docker-entrypoint.sh`

Reply:
Thanks for the clarification. Reconfigure the environment `LD_PRELOAD` in script would align the behavior. However, once running outside the container, the location of `libjemalloc.so` would be totally different in different versions of linux environment. To not extend this scope of this PR (mainly focus on the container environment), I think we would not pay too much efforts to find the location of  `libjemalloc.so`. The improvement could be done for others who runn Celeborn in non-container environments.

In a nutshell, I think moving the setting of `LD_PRELOAD` to environment related scripts would be a good idea, but we will only detect the `/usr/lib/$(uname -m)-linux-gnu/libjemalloc.so` location in this PR.

Reply:
We can simply consider the Ubuntu/CentOS in the detection logic, and allow the user to overwrite the `LD_PRELOAD` in `conf/celeborn-env.sh` in an arbitrary location.

Reply:
@mddxhj could you make such changes if you don't have further concerns?

Reply:
@pan3793 I think there's no other concerns，I'll try to migrate LD_PRELOAD configuration to load-celeborn-env.sh。

Reply:
> @mddxhj could you make such changes if you don't have further concerns?

@pan3793 I have changed the implementation as mentioned above.